base_iap_alternative/__manifest__.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2020 Akretion (https://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+{
+    "name": "Base IAP Alternative",
+    "summary": "Base module for providing alternative provider for iap apps",
+    "version": "12.0.1.0.0",
+    "category": "Tools",
+    "website": "http://github.com/OCA/server-tools",
+    "author": "Akretion, Odoo Community Association (OCA)",
+    "maintainers": ["sebastienbeau"],
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    "external_dependencies": {"python": [], "bin": []},
+    "depends": ["iap", "server_environment"],
```

Comment:
You mustn't depend on `server_environment`, as a lot of us don't use it as deployment method and this is forcing a deployment style.

Reply:
Thanks for you review Pedro but on this point I do not agree (let's explain my point of view)

In V12 @guewen did a huge work to make server_environment less invasive. 
Before V12 adding server_environment was forcing you to use files with password/config for the deployment. Like you I really dislike this ;) as it force to use this file and make the deployment more complex for a lot of integrator. We all have our own recipe and we need to take care of new comer.

But since the v12 (and thanks a lot @guewen for this). Now file for server_environment are optional !
If you do not have it, nothing change for you and by default the password/config are stored in the database.

The huge main point of this change is now module can directly depend of server_environment and this avoid the creation of a lot of glue module.
In my case I will need a "base glue module" then a glue module for all provider implementation (so we just duplicate the number of module by 2 ! each module will only have 5 lines of code...)

Moreover we also succeed to converge for the encryption of password in database per environment with server_environment_data_encryption https://github.com/OCA/server-env/tree/12.0/server_environment_data_encryption.
So now when a module is compatible with server_environment the enduser is able to configure the login/password for the different environment and then the database is duplicated into a different env you use the right configuration.

In my opinion storing password in clear in a database is a bad practice as it always generate issue when restoring a backup for testing (a duplicated database for testing interact with a service in production).
OCA should encourage good practice, server_environment is a really good module and we should encourage anyone to use it by default. Them as we all have our own recipe we can implement our own way (C2C and Acsone use file, on our side we use encrypted config per env), in any case if you do anything server_environment will just store it in clear like native behavior.

Interested to have your point of view (and also have other Pov like @guewen )

Reply:
As said, this is your deployment way. We don't need this module for anything, and I don't store password in DBs in clear for any deployment I do, and I don't want to depend on this module for everything, but I don't really plan to use as well this iap implementation for now. It was just an observation about the increase dependency of all your work in this module that I personally don't like.

Reply:
Actually, we use an environment variable, stored in a file in clear text for the non-sensitive config (SERVER_ENV_CONFIG) and in an encrypted file for sensitive config (SERVER_ENV_CONFIG_SECRET). These variables are set on the containers so a restart is enough to change an option (using files need a rebuild of image).

But I agree with @sebastienbeau even if you don't use server_environment, it won't change anything for you (if no value is set in the server env inputs (files, env variables), the fields are editable as usual.

For a module that mainly stores credentials, it makes sense to depends on it, furthermore as it doesn't force you to use its method.